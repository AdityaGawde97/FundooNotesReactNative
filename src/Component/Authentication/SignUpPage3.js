import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { styles } from '../../Css/Authentication.style'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button, CheckBox } from 'react-native-elements';
import { registerUser } from "../../Firebase/AuthServices";

export default class SignUpPage2 extends Component {
    constructor(props) {
        super(props);

        this.state = {
            password: '',
            confirm: '',
            visibleIcon: false,
            errors: {}
        };
    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;

        if (typeof this.state.password !== "undefined") {
            if (!this.state.password.match(/^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).*$/)) {
                formIsValid = false;
                errors["password"] = "*Enter secure and strong password.";
            }
        }

        if (!this.state.password) {
            errors["password"] = "*Enter password";
            formIsValid = false;
        }

        if (typeof this.state.confirm !== "undefined") {
            if (this.state.confirm !== this.state.password) {
                formIsValid = false;
                errors["confirm"] = "*Password does not match.";
            }
        }

        if (!this.state.confirm) {
            errors["confirm"] = "*Enter password";
            formIsValid = false;
        }


        this.setState({
            errors: errors
        })

        return formIsValid
    }

    submitUserSignUp = () => {
        if (this.validateForm()) {
            const { navigation } = this.props
            registerUser(navigation.getParam('firstName'),
                navigation.getParam('lastName'),
                navigation.getParam('email'),
                this.state.password,
                () => {
                    this.props.navigation.navigate('SignIn')
                }
            )
        }
    }

    render() {

        return (
            <View style={styles.mainContainer}>

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Create a strong password
                </Text>

                <Text
                    style={{
                        flex: 0,
                        textAlign: 'center',
                        fontSize: 15,
                        marginLeft: '8%',
                        marginRight: '8%',
                        top: '7%'
                    }}>
                    Create a strong password with a mix of letters, numbers and symbols
                </Text>



                <View style={styles.inputContainer}>

                    <View style={{ top: 10 }}>
                        <OutlinedTextField
                            testID="testPassword2"
                            label="Password"
                            value={this.state.password}
                            onChangeText={(password) => this.setState({ password })}
                            secureTextEntry={!this.state.visibleIcon}
                            error={this.state.errors.password}
                            errorColor='red'
                        />
                    </View>

                    <View style={{top: 20}}>
                        <OutlinedTextField
                            testID="testConfirmPassword"
                            label="Confirm"
                            value={this.state.confirm}
                            onChangeText={(confirm) => this.setState({ confirm })}
                            secureTextEntry={!this.state.visibleIcon}
                            error={this.state.errors.confirm}
                            errorColor='red'
                        />
                    </View>
                    <View style={{top: 10}}>
                        <CheckBox
                            testID="testPassVisibility2"
                            title="Show password"
                            checked={this.state.visibleIcon}
                            onPress={() => this.setState({ visibleIcon: !this.state.visibleIcon })}
                            containerStyle={{
                                backgroundColor: '#fff',
                                borderColor: '#fff',
                                right: 7
                            }}
                        />
                    </View>

                </View>

                <View style={styles.nextInSignUp}>
                    <Button
                        testID="testNextButton3"
                        title="Next"
                        onPress={this.submitUserSignUp}
                    />
                </View>

            </View>
        );
    }
}