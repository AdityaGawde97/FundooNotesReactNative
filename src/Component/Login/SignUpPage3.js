import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { styles } from '../../Css/LoginCss'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button } from 'react-native-elements';
import { registerUser } from "../../Firebase/FirebaseLoginServices";

export default class SignUpPage2 extends Component {
    constructor(props) {
        super(props);

        this.state = {
            password: '',
            confirm: '',
            visibleIcon: false,
            errors: {}
        };
    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;

        if (typeof this.state.password !== "undefined") {
            if (!this.state.password.match(/^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).*$/)) {
                formIsValid = false;
                errors["password"] = "*Enter secure and strong password.";
            }
        }

        if (!this.state.password) {
            errors["password"] = "*Enter password";
            formIsValid = false;
        }

        if (typeof this.state.confirm !== "undefined") {
            if (this.state.confirm !== this.state.password) {
                formIsValid = false;
                errors["confirm"] = "*Password does not match.";
            }
        }

        if (!this.state.confirm) {
            errors["confirm"] = "*Enter password";
            formIsValid = false;
        }


        this.setState({
            errors: errors
        })

        return formIsValid
    }

    submitUserSignUp = () => {
        if (this.validateForm()) {
            const { navigation } = this.props
            registerUser(navigation.getParam('firstName'),
                navigation.getParam('lastName'),
                navigation.getParam('email'),
                this.state.password,
                () => {
                    this.props.navigation.navigate('SignIn')
                }
            )
        }
    }

    render() {

        return (
            <View style={styles.mainContainer}>

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Create a strong password
                </Text>

                <Text
                    style={{
                        flex: 0,
                        textAlign: 'center',
                        fontSize: 15,
                        marginLeft: '8%',
                        marginRight: '8%',
                        top: '7%'
                    }}>
                    Create a strong password with a mix of letters, numbers and symbols
                </Text>



                <View style={styles.inputContainer}>

                    <View style={styles.passView}>
                        <OutlinedTextField
                            testID="testPassword2"
                            label="Password"
                            value={this.state.password}
                            onChangeText={(password) => this.setState({ password })}
                            secureTextEntry={!this.state.visibleIcon}
                            error={this.state.errors.password}
                            errorColor='red'
                        />
                    </View>

                    <View style={styles.passIconView}>

                        <View style={styles.passView2}>
                            <OutlinedTextField
                            testID="testConfirmPassword"
                                label="Confirm"
                                value={this.state.confirm}
                                onChangeText={(confirm) => this.setState({ confirm })}
                                secureTextEntry={!this.state.visibleIcon}
                                error={this.state.errors.confirm}
                                errorColor='red'
                            />
                        </View>
                        <View style={{ padding: '3%' }}>
                            <TouchableOpacity 
                                testID="testPassIcon2"
                                onPress={() => this.setState({ visibleIcon: !this.state.visibleIcon })}
                            >
                                {
                                    !this.state.visibleIcon ?
                                        <Image
                                            style={styles.passIcon}
                                            source={require('../../Assets/visible.png')}
                                        /> :
                                        <Image
                                            style={styles.passIcon}
                                            source={require('../../Assets/off.png')}
                                        />
                                }

                            </TouchableOpacity>
                        </View>
                    </View>
                </View>

                <View style={styles.nextInSignUp}>
                    <Button
                        testID="testNextButton3"
                        title="Next"
                        onPress={this.submitUserSignUp}
                    />
                </View>

            </View>
        );
    }
}