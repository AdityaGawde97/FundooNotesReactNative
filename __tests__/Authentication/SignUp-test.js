import React from 'react';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';
import SignUpPage1 from '../../src/Component/Authentication/SignUpPage1';
import SignUpPage2 from '../../src/Component/Authentication/SignUpPage2';
import SignUpPage3 from "../../src/Component/Authentication/SignUpPage3"

const findElement = (tree, element) => {
    let result = undefined;
    for(node in tree.children){
        if(tree.children[node].props.testId=element){
            result = true;
        }
    }
    return result;
}

/**************************** SignUpPage1 Component Testing ******************************/

const wrapper1 = shallow(<SignUpPage1/>)

// This test just uses Jest snapshot testing
it('SignUpPage1 Page renders correctly', () => {
    renderer.create(<SignUpPage1 />);
});

// Using Jest + Enzyme
describe('<SignUpPage1/>', () => {
    it('SignUpPage1 Matching SnapShot', () => {
        expect(shallow(<SignUpPage1 />)).toMatchSnapshot();
    });

    it('checking initial state', () => {
        expect(wrapper1.state('firstName')).toEqual('');
        expect(wrapper1.state('lastName')).toEqual('');
    })
    const firstName = wrapper1.find('OutlinedTextField').first();
    const lastName = wrapper1.find('OutlinedTextField').last();

    it('testing state of component after onChangeText event occur', () => {

        firstName.props().onChangeText('Aditya');

        lastName.props().onChangeText('Gawde');

        expect(wrapper1.state('firstName')).toEqual('Aditya');
        expect(wrapper1.state('lastName')).toEqual('Gawde');
    })

    it('props of OutlinedTextField', () => {
        expect(firstName.findWhere((node) => node.prop('label') === 'First name')).toExist();
        expect(lastName.findWhere((node) => node.prop('label') === 'Last name')).toExist();
    });

    it('testing validateForm mathod for first name and last name validation', () => {

        const instance = wrapper1.instance()
        expect(instance.validateForm()).toBe(true)

        firstName.props().onChangeText('');
        expect(instance.validateForm()).toBe(false)

        lastName.props().onChangeText('');
        expect(instance.validateForm()).toBe(false)

    })
});

/**************************** SignUpPage2 Component Testing ******************************/

const wrapper2 = shallow(<SignUpPage2/>)

it('SignUpPage2 Page renders correctly', () => {
    renderer.create(<SignUpPage2 />);
});

describe('<SignUpPage2/>', () => {
    it('SignUpPage2 Matching SnapShot', () => {
        expect(shallow(<SignUpPage2 />)).toMatchSnapshot();
    });

    it('checking initial state', () => {
        expect(wrapper2.state('email')).toEqual('');
    })
    const emailInput = wrapper2.find('OutlinedTextField').first();

    it('testing state of component after onChangeText event occur', () => {

        emailInput.props().onChangeText('andy@gmail.com');

        expect(wrapper2.state('email')).toEqual('andy@gmail.com');
    })

    it('props of OutlinedTextField', () => {
        expect(emailInput.findWhere((node) => node.prop('label') === 'Email')).toExist();
        expect(emailInput.findWhere((node) => node.prop('errorColor') === 'red')).toExist();
    });

    it('testing validateForm mathod for email validation', () => {

        const instance = wrapper2.instance()
        expect(instance.validateForm()).toBe(true)

        emailInput.props().onChangeText('andygmail.com');
        expect(instance.validateForm()).toBe(false)

        emailInput.props().onChangeText('andy@gmail.com');
        expect(instance.validateForm()).toBe(true)

    })
});

/**************************** SignUpPage3 Component Testing ******************************/

const wrapper3 = shallow(<SignUpPage3/>)
const tree3 = renderer.create(<SignUpPage3 />).toJSON()

it('SignUpPage3 Page renders correctly', () => {
    renderer.create(<SignUpPage3 />);
});

describe('<SignUpPage3/>', () => {
    it('SignUpPage3 Matching SnapShot', () => {
        expect(shallow(<SignUpPage3 />)).toMatchSnapshot();
    });

    it('checking initial state', () => {
        expect(wrapper3.state('password')).toEqual('');
        expect(wrapper3.state('confirm')).toEqual('');
    })

    const password = wrapper3.find('OutlinedTextField').first();
    const confirm = wrapper3.find('OutlinedTextField').last();

    it('testing state of component after onChangeText event occur', () => {

        password.props().onChangeText('Abc@1234');

        confirm.props().onChangeText('Abc@1234');

        expect(wrapper3.state('password')).toEqual('Abc@1234');
        expect(wrapper3.state('confirm')).toEqual('Abc@1234');
    })

    it('props of OutlinedTextField', () => {
        expect(password.findWhere((node) => node.prop('label') === 'Password')).toExist();
        expect(password.findWhere((node) => node.prop('errorColor') === 'red')).toExist();
        expect(confirm.findWhere((node) => node.prop('label') === 'Confirm')).toExist();
        expect(confirm.findWhere((node) => node.prop('errorColor') === 'red')).toExist();
    });

    it('testing validateForm mathod for email validation', () => {

        const instance = wrapper3.instance()
        expect(instance.validateForm()).toBe(true)

        password.props().onChangeText('Abc1234');
        expect(instance.validateForm()).toBe(false)
        password.props().onChangeText('abc@dgdfhd');
        expect(instance.validateForm()).toBe(false)
        password.props().onChangeText('abc1234');
        expect(instance.validateForm()).toBe(false)

        password.props().onChangeText('Abc@1234');
        expect(instance.validateForm()).toBe(true)

        confirm.props().onChangeText('Xyz@1234');
        expect(instance.validateForm()).toBe(false)

        confirm.props().onChangeText('Abc@1234');
        expect(instance.validateForm()).toBe(true)

    })

    it('testing visible password toggling', () => {
        const showPasswordBtn = wrapper.findWhere((node)=>node.prop('testId') === 'testPassVisibility2')
        expect(wrapper3.state('visibleIcon')).toEqual(false)
        showPasswordBtn.simulate('press');
        expect(wrapper3.state('visibleIcon')).toEqual(true)
        showPasswordBtn.simulate('press');
        expect(wrapper3.state('visibleIcon')).toEqual(false)
        expect(findElement(tree3, 'testPassVisibility2')).toBeDefined();
    })
});
