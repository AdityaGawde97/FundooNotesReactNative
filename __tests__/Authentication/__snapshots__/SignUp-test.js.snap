// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage1/> SignUpPage1 Matching SnapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontFamily": "Roboto",
        "fontSize": 26,
        "top": "4%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      F
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      u
    </Text>
    <Text
      style={
        Object {
          "color": "#F4B400",
        }
      }
    >
      n
    </Text>
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      d
    </Text>
    <Text
      style={
        Object {
          "color": "#0F9D58",
        }
      }
    >
      o
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      o
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 24,
        "top": "5%",
      }
    }
  >
    Create a Fundoo Account
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 18,
        "top": "6%",
      }
    }
  >
    Enter your name
  </Text>
  <View
    style={
      Object {
        "flex": 9,
        "flexDirection": "column",
        "margin": "8%",
        "top": "5%",
      }
    }
  >
    <View>
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="First name"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        testID="testFirstName"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "top": "3%",
        }
      }
    >
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Last name"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        testID="testLastName"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "flex": 1,
        "padding": "6%",
        "width": 180,
      }
    }
  >
    <ForwardRef(Themed.Button)
      onPress={[Function]}
      testID="testNextButton1"
      title="Next"
    />
  </View>
</View>
`;

exports[`<SignUpPage2/> SignUpPage2 Matching SnapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontFamily": "Roboto",
        "fontSize": 26,
        "top": "4%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      F
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      u
    </Text>
    <Text
      style={
        Object {
          "color": "#F4B400",
        }
      }
    >
      n
    </Text>
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      d
    </Text>
    <Text
      style={
        Object {
          "color": "#0F9D58",
        }
      }
    >
      o
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      o
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 24,
        "top": "5%",
      }
    }
  >
    Choose your Fundoo address
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 18,
        "top": "6%",
      }
    }
  >
    Create your own
  </Text>
  <View
    style={
      Object {
        "flex": 9,
        "flexDirection": "column",
        "margin": "8%",
        "top": "5%",
      }
    }
  >
    <View>
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Email"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        testID="testEmail2"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "flex": 1,
        "padding": "6%",
        "width": 180,
      }
    }
  >
    <ForwardRef(Themed.Button)
      onPress={[Function]}
      testID="testNextButton2"
      title="Next"
    />
  </View>
</View>
`;

exports[`<SignUpPage3/> SignUpPage3 Matching SnapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontFamily": "Roboto",
        "fontSize": 26,
        "top": "4%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      F
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      u
    </Text>
    <Text
      style={
        Object {
          "color": "#F4B400",
        }
      }
    >
      n
    </Text>
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      d
    </Text>
    <Text
      style={
        Object {
          "color": "#0F9D58",
        }
      }
    >
      o
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      o
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 24,
        "top": "5%",
      }
    }
  >
    Create a strong password
  </Text>
  <Text
    style={
      Object {
        "flex": 0,
        "fontSize": 15,
        "marginLeft": "8%",
        "marginRight": "8%",
        "textAlign": "center",
        "top": "7%",
      }
    }
  >
    Create a strong password with a mix of letters, numbers and symbols
  </Text>
  <View
    style={
      Object {
        "flex": 9,
        "flexDirection": "column",
        "margin": "8%",
        "top": "5%",
      }
    }
  >
    <View
      style={
        Object {
          "top": 10,
        }
      }
    >
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Password"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        secureTextEntry={true}
        testID="testPassword2"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "top": 20,
        }
      }
    >
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Confirm"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        secureTextEntry={true}
        testID="testConfirmPassword"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "top": 10,
        }
      }
    >
      <Themed.CheckBox
        checked={false}
        containerStyle={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#fff",
            "right": 7,
          }
        }
        onPress={[Function]}
        testID="testPassVisibility2"
        title="Show password"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "flex": 1,
        "padding": "6%",
        "width": 180,
      }
    }
  >
    <ForwardRef(Themed.Button)
      onPress={[Function]}
      testID="testNextButton3"
      title="Next"
    />
  </View>
</View>
`;
